### PR TITLE
Delivering all outputs

### PR DESCRIPTION
**Pull request by omokaro-faith**
What does this PR do?

Create output for solving the problem
Create output for writing test
Create output for using design pattern
Create output for setting up software development environment

Description of Task to be completed?
Create output for objective 6

How should this be manually tested?
Navigate to the output folder then view files

Any background context you want to provide?

NA

What are the relevant pivotal tracker stories?

NA

Screenshots (if appropriate)

NA

Questions:

NA

### CODE REVIEW

File: 06-building-a-program-start-to-finish/automating-tasks/Output/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: Gulp
+- **Deliver**: a screencast showing gulp watching and refreshing your browser when changes are made to your HTML, CSS & JS files
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[link to screencast showing gulp watching HTML, CSS & JS files !](https://www.youtube.com/watch?v=QxiJnVfEKkQ)
+
+### Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/automating-tasks/Output/README.md:10]: Ok

File: 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#### Output
+- **Using**: `Code Climate` and `Coveralls`
+- **Deliver**: a screenshot of your code coverage and GPA badges in your repo's README file
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[link to travis and coverage badge](https://github.com/andela-fomokaro/Inverted-Index-Project/blob/develop/Pictures%20of%20eslint%20error%20violation%20and%20houndci/travis%20and%20coverage%20badges.png)
+
+### Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/checking-repository-health/Output/README.md:10]: I see your badge.

File: 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Google Doc
+- **Deliver**: `Release notes` for your final product in your repo's wiki
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[Link to release note!](https://docs.google.com/a/andela.com/document/d/1kbvLKewvndFRu_7ez4qlVHpyv6Kun1CzDr0X-WlOiuw/edit?usp=sharing)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/delivering-your-product/Output/README.md:7]: Cool

File: 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+- **Deliver**: a post reflecting on your learning experience while producing these outputs
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+#### Output
+
+[Link to Medium Post!](https://medium.com/@faith.omokaro/checkpoints-and-simulations-my-road-to-learning-38c3ccf07c09#.e27qrpifq)
+
+## Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/reflecting-on-the-learning-experience/Output/README.md:12]: Seen. Good read.

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+#### Output
+- **Using**: Heroku
+- **Deliver**: live URLs to a `review` app, your `staging` & `production` apps
+- **With**: *independently*
+
+#### Targeted Learning Outcomes
+[Live URL to deployment on production application](https://fomokaro-inverted-production.herokuapp.com/) and [link to heroku application deployment on staging](https://fomokaro-invertedindex-staging.herokuapp.com/) and [link to review application](https://fomokaro-invertedindex-st-pr-2.herokuapp.com/)
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: Live links. But I believe you can format this ReadMe file better than this.

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+OUTPUT
+
+Link to Github repo
+https://github.com/andela-fomokaro/Inverted-Index-Project
+
+## Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md:11]: This readme file can be properly formatted for this output

File: 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+MVC is an achitectural pattern that is used to separate an application's concerns. That is, application code is broken into three layers:
+
+Model - represents an object carrying data. e.g. InvertedIndex Class
+View - represents the front end view of the data that model contains. e.g. index.html
+Controller - acts on both model and view. It controls the data flow into model object and updates the view whenever data changes. e.g. app.js
+Hence as explained above, in the case of the Inverted Index App, the model is the InvertedIndex Class, the view is the front end HTML and CSS, while the controller is the angular app.
+
+
+## Author
+### Omokaro Faith
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/using-design-patterns/Output/README.md:85]: Good.

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 8]
```diff
@@ -0,0 +1,174 @@
+const Index = require('../Model/inverted-index.js');
+const invalidBook = require('../files/test.json');
+const workBook = require('../files/work.json');
+const empty = require('../files/empty.json');
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:5]: A base filename should exactly match the name of its default export.  [See link](https://github.com/airbnb/javascript#naming--filename-matches-export)

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+            status: false,
+            msg: 'Invalid file content',
+          };
+        }
+      });
+    }
+    return check;
+    // } catch (error) {
+    //   return error.msg;
+    // }
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Why do you have commented code here?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+  }
+/**
+     * It returns the index of the file passed
+     *
+     * @method createIndex
+     * @param {Object} fileName
+     * @param {Object} jsonObject
+     *
+     * @return {number} returns the position on the file
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Are you returning a number?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+     */
+  createIndex(fileName, jsonObject) {
+    const newIndex = {};
+    // this.validateFile(jsonObject);
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Why is this here?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+    // } catch (error) {
+    //   return error.msg;
+    // }
+  }
+/**
+     * It returns the index of the file passed
+     *
+     * @method createIndex
+     * @param {Object} fileName
+     * @param {Object} jsonObject
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Is **_jsonObject_** an appropriate parameter name?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+/**
+     * it returns the word searched for in the object it was found
+     *
+     * @method searchIndex
+     * @param {String} sTerms
+     * @param {String} fileName
+     *
+     * @returns {Object} search result
+     */
+  searchIndex(sTerms, fileName) {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Whats is your basis for abbreviating your parameter name? Abbreviations do not enhance readability.

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+            }
+          });
+        }
+      });
+      return searchResult;
+    }
+    Object.keys(this.index).forEach((filename) => {
+      const selectedIndex = this.index[filename];
+      let terms = this.tokenize(sTerms);
+      // this.flattenSearch(terms);
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Why is this here?

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+     * @method searchIndex
+     * @param {String} sTerms
+     * @param {String} fileName
+     *
+     * @returns {Object} search result
+     */
+  searchIndex(sTerms, fileName) {
+    const searchResult = {};
+    const allIndex = {};
+    if (fileName !== 'Select file') {
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: If this condition is not fulfilled what happens?

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+describe('Search Index', () => {
+  it('Should return correct index of the search term in work.json', () => {
+    expect(invertedIndex.searchIndex('Testing, soon', 'work.json')).toEqual({
+      testing: [0],
+      soon: [1],
+    });
+  });
+  it('should return true if search term is a string', () => {
+    const words = 'Alice in Wonderland';
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:99]: Is this variable "words" used anywhere on this file?

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 17]
```diff
@@ -0,0 +1,174 @@
...
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.searchIndex).toBe('function');
+  });
+
+  it('should check that the class has a searchIndex method', () => {
+    expect(typeof invertedIndex.flattenSearch).toBe('function');
+  });
+});
+
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:174]: Your tests in general should use "Ubiquitous Language," 

Which is a body of knowledge that can be understood by both the developer and the customer. This ubiquitous language is used to shape and develop the requirements and testing needed, at the level of the customer's understanding.

```
describe("Calculator: add", function (){

    it("should be able to add 2 numbers together", function (){
        var total = add(1, 2);
        expect(total).toBe(3);
    });

    it("should be able to add 3 numbers together", function (){
        var total = add(1, 2, 3);
        expect(total).toBe(6);
    });
});
```

**Review by andela-moe — COMMENTED** [review]
Please review comments

File: 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md (hunk 5 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/setting-up-software-development-environments-(deployment)/Output/README.md]: ok

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 9 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: I have fixed that issue

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 11 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: I fixed that already, forgot to remove that after debugging

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 14 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: removed, that was an error

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 8 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:5]: I have made changes to my test file

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 16 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:99]: I have fixed the issue

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 17 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:174]: I fixed that

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 12 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: I have given the parameter a proper name

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 10 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: That was an error I have fixed that

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 13 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: I have changed the parameter name

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 15 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: Thanks I have refactored the code, It was an oversight

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 14 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js]: I have removed that , thanks

File: 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md (hunk 6 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/setting-up-your-repository/Output/README.md:11]: I have formatted the read file properly

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+          });
+        }
+      });
+      searchResult[filename] = temp;
+    });
+    return searchResult;
+  }
+}
+
+module.exports = InvertedIndex;
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:170]: I have given you feedback before on using abbreviations for your variables/parameter names. _ temp, obj_

Please you variable names/parameter names should be in camelCase not snake_case.

**Review by andela-moe — COMMENTED** [review]
Please address comments

**Comment by andela-moe** [comment]
You also seem to be missing a test for your readFile method

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 19]
```diff
@@ -0,0 +1,174 @@
...
+describe('Search Index', () => {
+  it('Should return correct index of the search term in work.json', () => {
+    expect(invertedIndex.searchIndex('Testing, soon', 'work.json')).toEqual({
+      testing: [0],
+      soon: [1],
+    });
+  });
+  it('should return true if search term is a string', () => {
+    const words = 'Alice in Wonderland';
+    expect(Object.keys(invertedIndex.searchIndex('words'))).toBeTruthy();
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:100]: Are you meant to pass a string directly here or **_words_** which you defined on line 99 of this file?

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js [hunk 20]
```diff
@@ -0,0 +1,174 @@
...
+    expect(invertedIndex.searchIndex).toBeDefined();
+  });
+});
+
+describe('Should return an object that is an index of the file', () => {
+  invertedIndex.createIndex('work.json', work);
+  it('should verify that index has been created', () => {
+    expect(Object.keys(invertedIndex.getIndex('work.json', work)).length)
+      .toBeGreaterThan(0);
+  });
```
Comment (andela-moe) [inline 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:43]: What exactly are you testing for in this describe block? Your getIndex() function?

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 17 above)
Comment (andela-moe) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:174]: I don't see your implementation of this feedback in your test

Your tests in general should use "Ubiquitous Language,"

File: 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js (hunk 18 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/solving-the-problem/Output/inverted-index.js:170]: I have made corrections

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 20 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:43]: I was actually testing to make sure the methods are function also, but I have made changes by removing the the block of code since it's not required

Comment (andela-moe) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:43]: Please put it back and make your Test Suite and test case clearer.

File: 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js (hunk 17 above)
Comment (omokaro-faith) [reply 06-building-a-program-start-to-finish/writing-tests/Output/spec/index-spec.js:174]: It was a mistake. I have made the correct changes now.
